data-prepper-plugin-framework/src/main/java/org/opensearch/dataprepper/plugin/PluginConfigurationConverter.java

```diff
@@ -92,4 +93,12 @@ private Object convertSettings(final Class<?> pluginConfigurationType, final Plu
         }
     }
 
+    private String constructConstrainViolationMessage(final ConstraintViolation<Object> constraintViolation) {
```

Comment:
We should probably split this into its own class since this will likely vary over time. Something like a `ConstraitViolationMessageProvider` class.